consensus/misc/create2deployer.go

```diff
@@ -40,10 +32,9 @@ func EnsureCreate2Deployer(c *params.ChainConfig, timestamp uint64, db vm.StateD
 	if !c.IsOptimism() || c.CanyonTime == nil || *c.CanyonTime != timestamp {
 		return
 	}
-	address, ok := create2DeployerAddresses[c.ChainID.Uint64()]
-	if !ok || db.GetCodeSize(address) > 0 {
+	if db.GetCodeSize(create2DeployerAddress) > 0 {
```

Comment:
I still lean towards always setting it because by putting this in the state transition function it becomes part of the protocol, so we should overwrite the deployed implementations that have an owner